pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -36,51 +37,56 @@ def __init__(
 
     def build_model(
         self,
-        data: pd.DataFrame,
-        adstock_max_lag: int = 4,
+        model_data: dict,
+        model_config: dict,
     ) -> None:
-        date_data = data[self.date_column]
-        target_data = data[self.target_column]
-        channel_data = data[self.channel_columns]
-        if self.control_columns is not None:
-            control_data: Optional[pd.DataFrame] = data[self.control_columns]
-        else:
-            control_data = None
-        coords: Dict[str, Any] = {
-            "date": date_data,
-            "channel": channel_data.columns,
-        }
-
-        if control_data is not None:
-            coords["control"] = control_data.columns
-
-        with pm.Model(coords=coords) as self.model:
+        with pm.Model(coords=model_config.coords) as self.model:
             channel_data_ = pm.MutableData(
                 name="channel_data",
-                value=channel_data,
-                dims=("date", "channel"),
+                value=model_data.channel_data,
+                dims=model_data.channel_data["dims"],
             )
 
-            target_ = pm.MutableData(name="target", value=target_data, dims="date")
+            target_ = pm.MutableData(
+                name="target",
+                value=model_data.target_data,
+                dims=model_config.target["dims"],
+            )
 
-            intercept = pm.Normal(name="intercept", mu=0, sigma=2)
+            intercept = pm.Normal(
+                name="intercept",
+                mu=model_config.intercept["mu"],
+                sigma=model_config.intercept["sigma"],
+            )
 
             beta_channel = pm.HalfNormal(
-                name="beta_channel", sigma=2, dims="channel"
```

Comment:
If these are hard-coded in the original model, we don't need to make them configurable here.

Reply:
what if I already added them? The basic functionality will stay the same, and the flexibility will be just in case we'd like to have it, but right now that would actually require work to make it have less flexibility